### PR TITLE
first batch of component

### PR DESCRIPTION
**Pull request by MarcoSantarelli**
**Related tasks:**
was added:
*  button component
*  checkbox component
*  input component
*  validation message
*  toast

**Before testing:**

1. Clone the PR 
2. run angular project

---

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+import {InputComponent} from './demo/input/input.component';
+import {CheckboxComponent} from './demo/checkbox/checkbox.component';
+import {GetStartedComponent} from './demo/get-started/get-started.component';
+import {IconComponent} from './demo/icon/icon.component';
 
 
-const routes: Routes = [];
+const routes: Routes = [
+    {path: 'getStarted', component: GetStartedComponent},
+    {path: 'buttons', component: ButtonDemoComponent},
```
Comment (danabalon) [inline src/app/app-routing.module.ts:13]: Should be ButtonComponent and the path 'button', to preserve consistency.

File: src/app/atoms/input/che-input.component.scss [hunk 2]
```diff
@@ -0,0 +1,164 @@
+
+/* form starting stylings ------------------------------- */
+.group 			  {
+  position:relative;
+}
+.inputMaterial 				{
```
Comment (danabalon) [inline src/app/atoms/input/che-input.component.scss:6]: change the name to che-input

File: src/app/atoms/input/che-input.component.scss [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  padding:10px 10px 10px 5px;
+  display:block;
+  width:300px;
+  border:none;
+  border-bottom:1px solid #757575;
+}
+.inputMaterial:focus 		{ outline:none;   }
+
+/* LABEL ======================================= */
+.labelMaterial 	{
```
Comment (danabalon) [inline src/app/atoms/input/che-input.component.scss:17]: change the name to che-label

**Review by danabalon — APPROVED** [review]

File: src/app/demo/button/button.component.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        <che-button color='primary' [outline]="true" label="Primary"></che-button>
+        <che-button color='secondary' [outline]="true" label="Secondary"></che-button>
+        <che-button color='success' [outline]="true" label="Success"></che-button>
+        <che-button color='warning' [outline]="true" label="Warning"></che-button>
+        <che-button color='danger' [outline]="true" label="Danger"></che-button>
+        <che-button color='info' [outline]="true" label="Info"></che-button>
+        <che-button color='dark' [outline]="true" label="Dark"></che-button>
+        <che-button color='ligth' [outline]="true" label="Ligth"></che-button>
+
+        <h5 class="mt-3">Buttons Sizes</h5>
```
Comment (aderenzis) [inline src/app/demo/button/button.component.html:59]: Button sizes

File: src/app/demo/button/button.component.html [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        <che-button color='primary' size="sm" label="Size Small"></che-button>
+        <che-button color='secondary' size="lg" label="Size Large"></che-button>
+        <h5 class="mt-3">Buttons Disabled</h5>
```
Comment (aderenzis) [inline src/app/demo/button/button.component.html:62]: Disabled buttons

File: src/app/demo/button/button.component.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            <che-icon type="light" icon="user"></che-icon>
+        </che-button>
+        <che-button color='ligth' label="Ligth">
+            <che-icon type="light" icon="user"></che-icon>
+        </che-button>
+
+        <h6 class="mt-2 sizeImportant">
+            <span class="text-info ">Important</span>: to implement "che-icon" you need install '@fidtech-sa/ft-icons' library before, <a routerLink="/icon">click here</a> for more information</h6>
+
+        <h5 class="mt-3">Severity Buttons Outline</h5>
```
Comment (aderenzis) [inline src/app/demo/button/button.component.html:49]: Outline severity buttons

File: src/app/demo/get-started/get-started.component.html [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        <h5><strong>Step 1</strong> - Install</h5>
+        <div class="containerBtnCopy">
+            <button (click)="copyText(codeInstall)" class="btn btn-outline-info" >Copy
+            </button>
+        </div>
+        <pre><code class="paddingCode" [highlight]="codeInstall"></code></pre>
+
+
+        <h5><strong>Step 2</strong> - Styles Config</h5>
+        <p>in angular.json add in "styles" the next line</p>
```
Comment (aderenzis) [inline src/app/demo/get-started/get-started.component.html:17]: in angular.json,  add in "styles" section the following line

File: src/app/demo/get-started/get-started.component.html [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        <div class="containerBtnCopy">
+            <button (click)="copyText(codeAngularJson)" class="btn btn-outline-info" >Copy
+            </button>
+        </div>
+        <pre><code class="paddingCode" [highlight]="codeAngularJson"></code></pre>
+
+        <h5><strong>Step 3</strong> - Config Variable of colors</h5>
```
Comment (aderenzis) [inline src/app/demo/get-started/get-started.component.html:24]: Color variables configuration

File: src/app/demo/get-started/get-started.component.html [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        <p>In styles.scss add the next code. in this code you can set color palet of your system</p>
```
Comment (aderenzis) [inline src/app/demo/get-started/get-started.component.html:25]: In styles.scss add the following code. This configures the color palet of your system

File: src/app/demo/icon/icon.component.html [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    <div class="col-12">
+        <h5><strong>Step 1</strong> - Install</h5>
+        <div class="containerBtnCopy">
+            <button (click)="copyText(codeInstall)" class="btn btn-outline-info">Copy
+            </button>
+        </div>
+        <pre><code class="paddingCode" [highlight]="codeInstall"></code></pre>
+
+
+        <h5><strong>Step 2</strong> - Styles and scripts Config</h5>
```
Comment (aderenzis) [inline src/app/demo/icon/icon.component.html:16]: Styles and scripts configuration

File: src/app/demo/icon/icon.component.html [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+        <p>in angular.json add in "styles" and "scripts" the nexts lines</p>
```
Comment (aderenzis) [inline src/app/demo/icon/icon.component.html:17]: In angular.json add in "styles" and "scripts" sections the following lines

**Review by aderenzis — APPROVED** [review]
All ok. Just a few comments about english improvements

**Pull request merged**